src/ripple/ledger/impl/PaymentSandbox.cpp

```diff
@@ -180,6 +213,19 @@ PaymentSandbox::balanceHook (AccountID const& account,
     return adjustedAmt;
 }
 
+std::uint32_t
+PaymentSandbox::ownerCountHook (AccountID const& account,
+    std::uint32_t count) const
+{
+    std::uint32_t result = count;
+    for (auto curSB = this; curSB; curSB = curSB->ps_)
+    {
+        if (auto adj = curSB->tab_.ownerCount (account))
+            result = std::max (result, *adj);
```

Comment:
Thanks for the explanation and for the comment on `ownerCountHook`.

So (I think) my final question on this topic is how we determine the `ownerCount` that is stored in all the accounts that had their offers consumed by the end of the payment (in the final ledger).  If we store the `ownerCount` that `ownerCountHook` is computing, then I'm worried we might be storing the wrong value.  I haven't internalized all of the magic associated with a `PaymentSandbox`.